### ISSUE TITLE
MAC Binding with DHCP Reservation and Firewall Configuration

### ISSUE THREAD

**Issue by Tbaile** [NethServer/nethsecurity#918]
MAC binding, also known as MAC address binding or MAC address reservation, is a network security feature that associates a specific MAC address with a particular network port or IP address. This ensures that only the device with the registered MAC address can use the designated network resource, enhancing security by preventing unauthorized devices from accessing the network.

**Description:**
- **Association between MAC address and IP:** Must be done using DHCP reservation.
- **Features:**
  - Enable/disable binding flag for each DHCP server.
  - Create the firewall configuration as a proof of concept (PoC) before developing the UI and API.

**Comment by gsanchietti** [comment]
Testing image:

#### Test Case 1: Enable Soft Binding
1. **Description**: Enable soft binding for the DHCP server and verify it is applied.
2. **Command**: 
   ```bash
   uci set dhcp.GREEN_1.ns_binding='1'
   uci commit dhcp
   reload_config
   ```
3. **Expected Result**: The soft binding should be enabled, enforcing the block only if the IP/MAC in the static leases mismatch.

#### Test Case 2: Enable Hard Binding
1. **Description**: Enable hard binding for the DHCP server and verify it is applied.
2. **Command**: 
   ```bash
   uci set dhcp.GREEN_1.ns_binding='2'
   uci commit dhcp
   reload_config
   ```
3. **Expected Result**: The hard binding should be enabled, enforcing the block if the client is not registered in the static leases.

#### Test Case 3: Disable Binding
1. **Description**: Disable the binding for the DHCP server and verify it is applied.
2. **Command**: 
   ```bash
   uci set dhcp.GREEN_1.ns_binding='0' # or uci delete dhcp.GREEN_1.ns_binding
   uci commit dhcp
   reload_config
   ```
3. **Expected Result**: The binding should be disabled, removing any binding enforcement.

#### Test Case 4: Verify Automatic Binding Updates
1. **Description**: Verify that IP/MAC bindings are automatically generated and updated when the DHCP configuration is changed.
2. **Command**: 
   ```bash
   uci set dhcp.GREEN_1.ns_binding='1'
   uci commit dhcp
   reload_config
   # Modify DHCP configuration here
   uci commit dhcp
   reload_config
   ```
3. **Expected Result**: The IP/MAC bindings should be updated according to the new DHCP configuration.

**Comment by gsanchietti** [comment]
All test cases have been verified.

**Comment by gsanchietti** [comment]
Released on 23.05.5-ns.1.4.0
